notebooks/demo_latticemaze.ipynb

```diff
@@ -40,6 +40,7 @@
                 "import matplotlib.pyplot as plt\n",
                 "\n",
                 "from maze_transformer.generation.generators import LatticeMazeGenerators\n",
+                "from maze_transformer.generation.lattice_maze import LatticeMaze, TargetedLatticeMaze, SolvedMaze\n",
                 "from maze_transformer.evaluation.plot_maze import MazePlot\n",
```

Comment:
There's probably a good reason for this, but what is the benefit of having three different classes over of having one lattice_maze class with optional variables of start_pos, end_pos, solution_path?